plonetheme/onegov/viewlets/searchbox.py

```diff
@@ -39,7 +39,11 @@ def placeholder(self):
         return getattr(self.context, 'search_label', default_placeholder)
 
     def get_form_tag_css_class(self):
+        # TODO: Should there be a class "no-solr" if there is no solr`. See the
+        # test `test_no_solr_cssclass_present_when_solr_disabled().
```

Comment:
What does this class do? If it is used somewhere in the solr package it can be skipped.

Reply:
I don't know what the classes `has-solr` and `no-solr` were used for, I cannot find any usage in `plonetheme.onegov`. The classes have been introduced in  https://github.com/OneGov/plonetheme.onegov/commit/ed9451f374c2b17c7e79c8e8ade8ebba3b6799b6 (`SearchBox: make placeholder customizable.`), but the addition of theses classes do not seem related to the topic `SearchBox: make placeholder customizable.` at all.

In https://github.com/OneGov/plonetheme.onegov/pull/215/commits/952c641da241ab1f661f53c70419a6a1918b0796 I made the searchbox viewlet compatible with Plone 5. Back then, I did not pay too much attention to these classes and I only noticed it when reworking the tests, so I added the `TODO`.

There is much code which requires `ftw.solr`. As I'm thinking about it right now, we should probably make `ftw.solr` compatible with Plone 5.1 very soon and tackle the whole searchbox topic again later.

I'm open for your further suggestions about the handling of this topic.

Reply:
The `has-solr` is used in `livesearch.js` (ex: https://github.com/4teamwork/plonetheme.teamraum/blob/master/plonetheme/teamraum/resources/js/livesearch.js#L43).
This is used in `plonetheme.teamraum`, `plonetheme.gever` and `opengever.core`. They all provide the `livesearch.js`.
Now `plonetheme.onegov` does not provide the `livesearch.js`. So my guess is that `has-solr` does exactly nothing and can be deleted.

Reply:
Thank you very much for your comments. I have removed the `has-solr` stuff.